src/SprykerEco/Zed/Heidelpay/Persistence/HeidelpayRepository.php

```diff
@@ -165,29 +164,17 @@ public function findHeidelpayDirectDebitRegistrationByIdAndTransactionId(
     }
 
     /**
-     * @param int $idSalesOrder
-     * @param string $transactionType
+     * @param \Generated\Shared\Transfer\PaymentHeidelpayTransactionLogCriteriaTransfer $paymentHeidelpayTransactionLogCriteriaTransfer
      *
-     * @return \Generated\Shared\Transfer\HeidelpayTransactionLogTransfer|null
+     * @return bool
      */
-    public function findHeidelpayTransactionLogByIdSalesOrderAndTransactionType(
-        int $idSalesOrder,
-        string $transactionType
-    ): ?HeidelpayTransactionLogTransfer {
-        $paymentHeidelpayTransactionLogEntity = $this->getPaymentHeidelpayTransactionLogQuery()
-            ->filterByFkSalesOrder($idSalesOrder)
-            ->filterByTransactionType($transactionType)
-            ->findOne();
-
-        if ($paymentHeidelpayTransactionLogEntity === null) {
-            return null;
-        }
-
-        return $this->getMapper()
-            ->mapEntityToHeidelpayTransactionLogTransfer(
-                $paymentHeidelpayTransactionLogEntity,
-                new HeidelpayTransactionLogTransfer(),
-            );
+    public function hasPaymentHeidelpayTransactionLog(PaymentHeidelpayTransactionLogCriteriaTransfer $paymentHeidelpayTransactionLogCriteriaTransfer): bool
+    {
+        return $this->getFactory()->createPaymentHeidelpayTransactionLogQuery()
```

Comment:
please create `setPaymentHeidelpayTransactionLogFilters` method similar to `\Spryker\Zed\ProductConfiguration\Persistence\ProductConfigurationRepository::setProductConfigurationFilters`